src/cpp/astar.cpp

```diff
@@ -73,6 +96,29 @@ static PyObject *astar(PyObject *self, PyObject *args) {
   nodes_to_visit.push(start_node);
 
   int* nbrs = new int[8];
+  
+  // Get the heuristic method to use
+  float (*heuristic_func)(int, int, int, int, int, int);
+  
+  if (heuristic_override == 1) {
+    heuristic_func = linf_norm;
+  } else if (heuristic_override == 2) {
+    heuristic_func = l1_norm;
+  } else if (heuristic_override == 3) {
+    heuristic_func = l1_orthogonal_x;
+  } else if (heuristic_override == 4) {
+    heuristic_func = l1_orthogonal_y;
+  } else {  // default
+    if (diag_ok) {
+      heuristic_func = linf_norm;
+    } else {
+      heuristic_func = l1_norm;
+    }
+  }
+  
+  // calculate max_x and max_y
+  int max_x = std::abs(goal / w - start / w);
+  int max_y = std::abs(goal % w - start % w);
```

Comment:
Didn't want to compute this in every heuristic call, I think we should leave it

Reply:
Modern compilers are very good at common subexpression elimination. At `O3` level of optimization I don't expect this to have any noticeable impact. (But if you have quantitative numbers showing otherwise I'd love to see them and understand why.)